bpf/estat/zpl.c

```diff
@@ -42,7 +46,7 @@ equal_to_pool(char *str)
 }
 
 static inline int
-zfs_read_write_entry(io_info_t *info, struct inode *ip, uio_t *uio, int flags)
+zfs_read_write_entry(io_info_t *info, struct inode *ip, zfs_uio_t *uio, int flags)
```

Comment:
Does this change work for all previous kernel versions? E.g. I'm curious about the case where a customer may upgrade, but continue running a kernel from a prior released (i.e. no reboot), would this script continue to work?

Reply:
No, that would be a problem.

We could deliver separate C scripts for each kernel version and then have the estat python script run the correct one.   Is that a problem we want to solve?

Reply:
> Is that a problem we want to solve?

I'd like to defer this question to the team.

Today, AFAIK, we do support deferred/non-reboot upgrades from any prior 6.0-based release to the latest one. So it's possible that we could have a system running the kernel bits from the 6.0.0.0 release, but the userland bits from the most recent 6.0.6.0 release.

In that case, I think these scripts would no longer work, since by design we only (currently) support running the scripts on the matching kernel for that release, right? So, we'd need to decide as a team, if it's OK for the scripts not to work in such a scenario.

If the current architecture of these scripts is to only work when run on the kernel version of the matching release (e.g. 6.0.6.0 scripts only work with the 6.0.6.0 kernel and modules), then I'll approve this, since it's consistent with our existing design decisions, even though I feel like that design is lacking and prone to failure on deferred upgrades.

Reply:
I agree, the current perf-diag design does not address deferred upgrade.   It is something we should give some thought too.  Changes to stbtrace scripts could be more problematic since they are used in analytics.   

It does seem to be outside the scope of this PR.   At a miniumum, the scripts should run on matching kernel versions.

Reply:
Brad, from my understanding those scripts are only run manually by support, is that right?

If that's the case then having them work for deferred upgrade is probably not a P1, but we should definitely create a bug and allocate time for it.

Reply:
While this specific script may only be run by support, as Brad mentioned, there's other scripts that are used by the product (for analytics) and may suffer from this same problem on deferred upgrade.

Reply:
Seb mentioned to me that we already have a bug tracking this.   I was just unsuccessful finding it but I'll make sure there is a jira issue.

Reply:
Yeah, for other scripts that are used by analytics, I think supporting deferred upgrade is a must.